File: db/migrate/formatData.js
const fs = require("fs");
const path = require("path");

const dataFile = fs.readFileSync(path.join(__dirname, "data.json"));

const data = JSON.parse(dataFile)["values"].slice(2);

for (let i in data) {
  const d = data[i];

  const name = d[5].replace(/('|’)/g, "''");
  const address = d[6].replace(/('|’)/g, "''");
  const city = d[8].replace(/('|’)/g, "''");
  const state = d[9].replace(/('|’)/g, "''");
  const instructions = d[10].replace(/('|’)/g, "''");
  const needs = d[11].replace(/('|’)/g, "''");
  const acceptsOpened = d[12].replace(/('|’)/g, "''");
  const latitude = d[13];
  const longitude = d[14];

  const insertOrg = `
INSERT INTO organizations
    (
      name,
      address,
      city,
      state,
      instructions,
      accepts_opened,
      latitude,
      longitude
    )
  VALUES
    (
      '${name}',
      '${address}',
      '${city}',
      '${state}',
      '${instructions}',
      '${acceptsOpened}',
      '${latitude}',
      '${longitude}'
    );`;

  const insertOrgNeeds = needs
    .split(",")
    .map(
      (need) =>
        `
INSERT INTO organization_needs
    (need, org_id)
  VALUES
    ('${need.trim()}',
    (SELECT id FROM organizations o
      WHERE o.name = '${name}'
        AND o.latitude = '${latitude}'
        AND o.longitude = '${longitude}'));`
    )
    .join("\n");

  const statements = `
BEGIN;
${insertOrg}
${insertOrgNeeds}

COMMIT;
`;

  fs.writeFileSync(
    path.join(__dirname, "..", "sql", "organizations", `${i}.sql`),
    statements
  );
}

File: db/migrate/format_data.py
"""
 This scripts generates sql/organization/*.sql files based on organizations found in data.json
"""

import json

with open('data.json') as f:
    DATA = json.load(f)['values']

    for d in DATA.len():
        name = d[5]
        address = d[6]
        city = d[8]
        state = d[9]
        instructions = d[10]
        needs = d[11]
        accepts_opened = d[12]
        latitude = d[13]

        insert = 'INSERT INTO '
        try:
            longitude = d[14]
        except Exception:
            longitude = ''
        res['rows'].append({
            'name': name,
            'address': address,
            'city': city,
            'state': state,
            'instructions': instructions,
            'needs': needs,
            'accepts_opened': accepts_opened,
            'latitude': latitude,
            'longitude': longitude
        })

        with open('{}.sql'.format('foo'), 'w') as outfile:
            json.dump(res, outfile)
